File: agentski-front/src/app/model/message.ts
import { User } from "./user";

export class Message{
    id: number; 
	createdTime: string; //"yyyy-MM-dd@HH:mm:ss"
    content: string;
    sender: User;
    receiver: User;
	
    constructor(){
        this.id = 0;
        var d = new Date();
        this.createdTime = ("0" + d.getDate()).slice(-2) + "-" + ("0"+(d.getMonth()+1)).slice(-2) + "-" +
        d.getFullYear() + " " + ("0" + d.getHours()).slice(-2) + ":" + ("0" + d.getMinutes()).slice(-2) + ":" + "00";
        this.content = "";
        this.sender = new User();
        this.receiver = new User();
    }
}

File: agentska-aplikacija/src/main/java/com/megatravel/agentskaaplikacija/model/Message.java
package com.megatravel.agentskaaplikacija.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.hibernate.annotations.CreationTimestamp;

@Entity
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
    @Column(nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
	private Date createdTime;
    @Column(nullable = false, unique = false, length = 512)
    private String content;
    @ManyToOne
    private User sender;
    @ManyToOne
    private User receiver;
    
    public Message() { }

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Date getCreatedTime() {
		return createdTime;
	}

	public void setCreatedTime(Date createdTime) {
		this.createdTime = createdTime;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public User getSender() {
		return sender;
	}

	public void setSender(User sender) {
		this.sender = sender;
	}

	public User getReceiver() {
		return receiver;
	}

	public void setReceiver(User receiver) {
		this.receiver = receiver;
	}
    
}
